CI: Improved K8s Kubeconformance validation

PhilipSchmid opened the pull request:
### Description
- Improved Helm chart validation by using Kubeconformance with enabled CR validation.
- Added JSON schemas for the Tetragon CRDs as we might need them in the future to validate CRs from the Helm chart (TPs, etc.).

Follow up PR of https://github.com/cilium/tetragon/pull/2804

### Changelog

```release-note
CI: Improved K8s Kubeconformance validation
```

PhilipSchmid commented:
Example runs / outputs:
- https://github.com/cilium/tetragon/actions/runs/10418288104/job/28854158754?pr=2811#step:10:1
- https://github.com/cilium/tetragon/actions/runs/10418288104/job/28854158754?pr=2811#step:11:1

lambdanis commented on install/kubernetes/Pipfile:
```diff
@@ -0,0 +1,12 @@
+[dev-packages]
+
+[requires]
+python_version = "3.12"
```
It seems that Renovate might not recognize it automatically https://github.com/renovatebot/renovate/issues/28899

I think it's fine, we can get back to it in some undefined future.

PhilipSchmid replied:
Let's give it a try to see if it works out of the box (I think it should be based on [this](https://docs.renovatebot.com/python/#sources-in-pipfile)), and if not, I'll address it later on.

PhilipSchmid commented on install/kubernetes/Makefile:
```diff
@@ -40,3 +44,28 @@ docs:
+# JSON schema of the TracingPolicy CRD. This target generates such schema.
+# Requires pipenv to be pre-installed.
+.PHONY: generate-jsonschemas
+generate-jsonschemas: $(CRDS)
```
Background information: I implemented this because the `ubuntu-latest` GH runner has `pyyaml` installed, but many OS (like macOS) don't. Also, it's not in Python's standard library. Hence, it would fail locally on many OSes. Furthermore, `pipenv` isn't installed on the `ubuntu-latest` GH runner, so if we would like to go with the pipenv route everywhere, we would need to install it in the GH actions.

I'm also fine with doing everything using `pipenv`.

PhilipSchmid commented on install/kubernetes/Makefile:
```diff
@@ -24,7 +29,6 @@ deps:
 .PHONY: lint
 lint:
 	$(HELM) lint . --with-subcharts
-	$(HELM) template tetragon . | docker run --rm -i $(KUBECONFORM_IMAGE) --strict --schema-location default
```
Valid point. Re-added them as target calls on the `all` target.

PhilipSchmid commented on .github/workflows/lint-helm.yaml:
```diff
@@ -29,3 +50,80 @@ jobs:
+        uses: mathiasvr/command-output@v2.0.0
+        with:
+          shell: bash
+          run: |
```
Makes sense. Addressed via the latest commit.

lambdanis commented on install/kubernetes/Makefile:
```diff
@@ -40,3 +45,35 @@ docs:
+	-f values.yaml \
+	--set crds.installMethod=helm \
+	--set tracingPolicies.default.enabled=true |\
+		kubeconform \
```
It seems we don't use `KUBECONFORM_IMAGE` (L6-7) anymore, so we could remove it, but maybe better run kubeconform as a container here? i.e. `docker run --rm -i $(KUBECONFORM_IMAGE)` - then we don't need to install kubeconform in CI extra.

PhilipSchmid replied:
Good point. I addressed that in the newest commit. I've tested it locally, but let's wait until GH CI finishes to see if everything works smoothly. If it does, we could ship it, in my opinion.

lambdanis reviewed (approved):
looks good, thank you @PhilipSchmid! I left two small comments, but approving anyway.

jrfastab commented:
@PhilipSchmid if you could address Anna's small comments then lets get this merged. Thanks.

PhilipSchmid commented:
@jrfastab I implemented Anna's feedback, squashed and rebased to main and force-pushed, and all CI tests were successful. IMO, this is ready to be merged 🚀.

lambdanis commented on install/kubernetes/Makefile:
```diff
@@ -40,3 +45,35 @@ docs:
+	$(HELM) template $(TETRAGON_CHART) . \
+	-f values.yaml \
+	--set crds.installMethod=helm \
+	--set tracingPolicies.default.enabled=true |\
```
nit: such value doesn't exist (yet?)

PhilipSchmid replied:
Refactored as discussed offline. See latest commit.

lambdanis reviewed (approved):
looks good, thanks!
